tracer/src/Datadog.Trace/Span.cs

```diff
@@ -154,22 +155,58 @@ public override string ToString()
         /// <param name="key">The tag's key.</param>
         /// <param name="value">The tag's value.</param>
         /// <returns>This span to allow method chaining.</returns>
-        internal ISpan SetTag(string key, string value)
+        internal Span SetTag(string key, string value)
```

Comment:
Is there a reason for this change? 🤔 We _explicitly_ converted it from returning `Span` to `ISpan` in [this PR](https://github.com/DataDog/dd-trace-dotnet/commit/83e2ce164904b48cb2a5c3246bb83abdebf4b295#diff-1e8df4e227e07ad44ab3b68c56caa5ff877ea6186f7c93cad43ed0c42ae71e94). I can't see why it would be an issue, but I know we do some funny stuff with tracer mismatch etc, so don't want to inadvertently affect anything with that...

Reply:
Ha, me trying to sneak in a tiny change. TBH, I saw it and I thought it was just something we missed in that PR you linked. There's no reason for the internal method to return `ISpan`, only the public API needs to do that.

I'll revert it since there's an obvious concern about version mismatch. Thanks. 😅

Reply:
Done.